dali/pipeline/data/tensor_vector.cc

```diff
@@ -507,26 +658,62 @@ void TensorVector<Backend>::Reset() {
 template <typename Backend>
 template <typename SrcBackend>
 void TensorVector<Backend>::Copy(const TensorList<SrcBackend> &in_tl, AccessOrder order) {
+  // This variant will be removed with the removal of TensorList.
   SetContiguous(true);
+
+  auto copy_order = copy_impl::SyncBefore(this->order(), in_tl.order(), order);
+
+
+  tl_->Resize(in_tl.shape(), in_tl.type());
+  type_ = in_tl.type_info();
+  sample_dim_ = in_tl.shape().sample_dim();
+
+  copy_impl::SyncAfterResize(this->order(), copy_order);
+
+  // Update the metadata
   type_ = in_tl.type_info();
   sample_dim_ = in_tl.shape().sample_dim();
-  tl_->Copy(in_tl, order);
+
+  // Here both batches are contiguous
+  type_info().template Copy<Backend, SrcBackend>(
+      unsafe_raw_mutable_data(*tl_), unsafe_raw_data(in_tl), in_tl.shape().num_elements(),
+      copy_order.stream(), false);
+  copy_impl::SyncAfter(this->order(), copy_order);
```

Comment:
What's the point of this exercise? Is it to get rid of `tl_->Copy` because there will be no more tl_ soon?

Reply:
Yes. Also, this overload will be gone, so I can probably keep it in the old state, I just used similar code in both variants. I can also call the `CopyImpl` here to hide the details if you prefer.

Reply:
reverted.

Reply:
On the second thought, I've got a bit too forward with how it will look.  As there will be intermediate step before removing TL, this change is useful as it prepares the code for the lack of TL as a member of TV